tests/integration/test_edge_cases.py

```diff
@@ -131,3 +132,26 @@ def test_convert_nwb_dataset_on_mock_datalad_dataset_with_broken_symlink(
     nwb2bids.testing.assert_subdirectory_structure(
         directory=temporary_bids_directory, expected_structure=expected_structure
     )
+
+
+def test_symlink_resolves_correctly_with_relative_path(
+    minimal_nwbfile_path: pathlib.Path, temporary_bids_directory: pathlib.Path
+):
+    """Test that symlinks created from relative paths resolve to the correct target."""
+    # Change to the parent directory and use a relative path to the NWB file
+    original_cwd = os.getcwd()
+    try:
```

Comment:
Could you use the `monkeypatch` approach instead of `os` control? It is a bit cleaner and shouldn't require the `try/finally` either

Example: https://github.com/con/nwb2bids/blob/main/tests/unit/test_directory_conditions.py#L23

Reply:
that is cleaner nice comment